module3/projects/rails_engine/requirements.md

```diff
@@ -135,115 +210,177 @@ Example JSON response for the Merchant resource:
   "data": {
     "id": "1",
     "type": "merchant",
     "attributes": {
-      "name": "Store Name"
+      "name": "New Widget Name",
+      "description": "High quality widget, now with more widgety-ness",
+      "unit_price": 299.99,
+      "created_at": "2020-12-18T13:42:56 UTC",
+      "updated_at": "2020-12-20T18:12:45 UTC"
     }
   }
 }
 ```
 
-### Destroy Record
+## RESTful: Destroy an Item
 
-This endpoint should destroy the corresponding record (and any associated data, eg if you delete a merchant you should delete their items too). Since this is a permanent destruction of data, we do not return any response body of any kind, and should send a 204 HTTP status code.
-
-*Extension*: when you delete an item, you will delete entries in the invoice_items table per the work above. This, however, might leave invoices in the database with no items at all, so you should find a way to clean those up, too.
+This endpoint should:
+* destroy the corresponding record (if found) and any associated data
+* destroy any invoice if this was the only item on an invoice
+* NOT return any JSON body at all, and should return a 204 HTTP status code
+* NOT utilize a Serializer (Rails will handle sending a 204 on its own if you just `.destroy` the object)
 
 
-## 3b. Relationships
+## RESTful: Relationship Endpoints
 
-These endpoints should show related records. The relationship endpoints you should expose are:
+These endpoints should show related records for a given resource. The relationship endpoints you should expose are:
 
 * `GET /api/v1/merchants/:id/items` - return all items associated with a merchant.
-* `GET /api/v1/items/:id/merchants` - return the merchant associated with an item
-
-## 3c. Find Endpoints
-
-In addition to the standard ReST endpoints, you will need to build "find" endpoints for both **Items** and **Merchants**.
-
-### Single Finders
-
-This endpoint should return a single record that matches a set of criteria. Criteria will be input through query parameters.
-
-The URI should follow this pattern: `GET /api/v1/<resource>/find?<attribute>=<value>`
-
-This endpoint should:
-
-* work for any attribute of the corresponding resource including the `updated_at` and `created_at` timestamps.
-* find partial matches for strings and be case insensitive, for example a request to `GET /api/v1/merchants/find?name=ring` would match a merchant with the name `Turing` and a merchant with the name `Ring World`.
-_Note: It does_ __NOT__ _need to accept multiple attributes, for example_ `GET /api/v1/items/find?name=pen&description=blue`
-
-Example JSON response for `GET /api/v1/merchants/find?name=ring`
+  * return a 404 if merchant is not found
+* `GET /api/v1/items/:id/merchant` - return the merchant associated with an item
+  * return a 404 if the item is not found
+  
+
+## Non-RESTful Search Endpoints
+
+In addition to the standard RESTful endpoints described above, you will build the following endpoints which will NOT follow RESTful convention:
```

Comment:
To be consistent with messaging, maybe update to `you will build 1 pair of the following endpoints'

Reply:
Yes, good catch